### DISCUSSION TITLE
Automated local controls (Bluecoin and other BLE sensors)

### DISCUSSION THREAD

**Discussion by revk** [revk/xESP32-Faikout#26] [General]
The Daikin module is designed to work with an external [Environmental monitor](https://github.com/revk/ESP32-EnvMon) allowing fine control of temperature using an external reference point.

The way it works involves telling the air-con a high or low target to make the target temperature stable around the reference temperature. It is usually much tighter control than just using the aircon's own temperature setting or "auto" mode.

Updates have been issued now to start providing some *local* automation. This means that the target can be set, with a tolerance, and the local controls try and keep the target at the specified temperature. I also plan to add local control for turning on and off at preset times.

<img src=https://user-images.githubusercontent.com/996983/232214593-779927db-2887-49d8-b9a9-1947cbe3a5c1.png align=right width=33%>

However, in addition, I have coded extra BLE tracking of BlueCoinT temperature sensors. The idea is you can use a remote temperature reference for the automated local controls. E.g. stick one of these by your bed, for example.

<img width="377" alt="Screenshot 2023-04-15 at 13 37 14" src="https://user-images.githubusercontent.com/996983/232224081-f7b96c62-bf0d-4483-a886-79bff3b15032.png">

<img src=https://github.com/revk/ESP32-Faikin/assets/996983/f8189a20-be52-4b7a-9eeb-a550a7b94e7a align=right width=33%>

Also, now, Telink sensors if refreshed.

**Comment by revk** [comment]
I have made the BLE based on a setting not `ble` as it seems to impact TCP and web controls. Obviously I am working on that anyway. But normal use with no `ble` will be unaffected.

**Comment by MaartenBremer** [comment]
Would this work with something like the XIAOMI Mijia LYWSD03MMC as well?

**Reply by revk** [reply]
I have no idea what that is. It does not sound like Daikin, so I doubt it would have a Daikin S21 lead or protocol.

**Reply by XMaarten** [reply]
Ìt is a BLE thermometer/hygrometer. that I would like to use instead of the BlueCoinT above.

**Reply by revk** [reply]
Ah sorry, misunderstood.

It might, but I have coded for the bluecoin one. Do you have a spec of its BLE advertisements at all?

**Reply by XMaarten** [reply]
The Xiaomi Mijia BLE sensors are supported by ESPHome as well, see [their code](https://github.com/esphome/esphome/blob/78cb0986914aa3763058f9f46175782ce4218bf2/esphome/components/xiaomi_lywsd03mmc/xiaomi_lywsd03mmc.cpp) . [This blog](https://blog.ostebaronen.dk/2019/09/mijia-ble.html) has more information about the data itself.

**Reply by revk** [reply]
Then I may be able to add. Where can I get one to play with?

**Reply by XMaarten** [reply]
They are just 10 pound on Amazon UK, but if you can let me know your address I can send you one as well.

**Reply by revk** [reply]
I don't suppose you have an amazon link (UK)?

**Reply by XMaarten** [reply]
https://www.amazon.co.uk/Mi-Temperature-Humidity-Monitor-2/dp/B08C7KVDJW/

**Reply by revk** [reply]
Wow, OK, that looks a lot like the Shelly one. I've ordered one to look in to.

**Reply by revk** [reply]
That blog post gives exactly what I need - it should not be hard to make it work. I'll let you know.

**Reply by XMaarten** [reply]
It is really small compared to the Shelly one, it is about the size of the BlueCoinT I think.

**Reply by revk** [reply]
Err - this?

![41OuM96dsGL _AC_SL1080_](https://github.com/revk/ESP32-Faikin/assets/996983/9e31150e-9de8-4c46-9bca-1cbeb9af693b)

**Reply by XMaarten** [reply]
43mm x 43 mm :)

**Reply by revk** [reply]
Hmm, and bluecoin is 36mm diameter, so yeh, small.

**Reply by revk** [reply]
Coming Saturday.

**Reply by revk** [reply]
Well, first off, it claims to be `LYWSD03MMC`, not `MJ_HT_V1` it seems... I'll keep at it.

**Reply by revk** [reply]
OK, it looks like it may need a config to make it send temp in advertisements. I'll look more.

**Reply by revk** [reply]
Worse, it seems the data is encrypted and one needs a bind key, which means this will not be a simple device to use with Faikin. Shame as they look great for it. I'll do a bit more research anyway.

**Reply by XMaarten** [reply]
Weird, the one I have does not use encryption. I think you have a newer firmware. At [ESPHome they describe how to get the bindkey](https://esphome.io/components/sensor/xiaomi_ble.html).

**Reply by revk** [reply]
Yeh, ideally we need to be able to detect and pair on the Faikin module, which means understanding the underlying BLE messages to achieve that - which is a lot more work. Not saying no, yet, but way more than simply recognising and decoding the advertisement like I can with the BlueCoinT.

**Reply by XMaarten** [reply]
Or maybe just use Esphome together with [MQTT ](https://esphome.io/components/mqtt.html) and read the temperature from MQTT instead.

**Reply by revk** [reply]
We could make the Faikin accent a reference temp via MQTT, yes. If anyone has a suitable set up and can confirm the MQTT topic/data, I can look at adding a setting for that.

**Reply by andreagdipaolo** [reply]
Do you still need that? I would like to use Zigbee thermometers as temperature references via Zigbee2mqtt. I can't quite get how to do that, if it even is possible at all.

**Reply by revk** [reply]
The manual defines some control messages we accept for external control and reference over MQTT.

**Reply by andreagdipaolo** [reply]
I've taken a look but it doesn't seem easy, or maybe I'm just not too familiar with MQTT. But basically how does it work? You make an override of the internal sensor or you just fiddle with the target temperature in order to reach the target on the sensor and adjust it dynamically?

**Reply by revk** [reply]
The faikin can't override the aircon sensor directly. So the faikin auto mode fiddles with the target to make it heat/cool to get it right.

With MQTT messages you can tell the faikin the target and the temp you are seeing. Do that at least once a minute. It will the cause the faikin to use that as a reference and drive the aircon accordingly.

Eg this is from my temp sensor, tells the faikin the temp is 20.68C now and I want the room 20.5C to 21C.

`command/AdrianAC/control {"env":20.68,"target":[20.500,21.000]}
`

**Comment by Elwell** [comment]
I take it you're aware if the alternative BThome compatible firmware for
these?

https://github.com/pvvx/ATC_MiThermometer

DWISOTT for all of mine

A

**Reply by revk** [reply]
Again, not a simple matter for people to just *use* them if they have to reflash first.

**Reply by revk** [reply]
That said, that is very clever stuff... Looking now.

**Reply by revk** [reply]
OK just followed https://github.com/pvvx/ATC_MiThermometer#flashing-or-updating-the-firmware-ota

So far, I am impressed...

**Reply by revk** [reply]
Ok That seems actually not to hard, so yeh, I'll keep working.

**Reply by revk** [reply]
OK new code issued - re-flash the module as per instructions and should appear by name of MAC address. Have fun.
